### PR TITLE
Fix horovod installation `base-cuda` Dockerfile

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes a similar problem to #11596, but the issue is horovod this time.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @tchaton @rohitgr7 @akihironitta @carmocca @borda

### CODE REVIEW

**Comment by akihironitta** [comment]
Watching if `pip list` in `Docker / build-CUDA (3.7, 1.8) (pull_request)` shows `horovod`.
https://github.com/PyTorchLightning/pytorch-lightning/runs/5112169116?check_suite_focus=true

**Comment by Borda** [comment]
@akihironitta very nice, just adding sanity checks that the installed packages are available (can be imported)

File: dockers/base-cuda/Dockerfile [hunk 1]
```diff
@@ -108,15 +108,18 @@ RUN \
 
 RUN \
     # install FairScale
-    pip install fairscale==0.4.0
+    pip install fairscale==0.4.0 && \
+    python -c "import fairscale ; print(fairscale.__version__)"
```
Comment (carmocca) [inline dockers/base-cuda/Dockerfile:112]:
```suggestion
    python -c "import fairscale; print(fairscale.__version__)"
```

File: dockers/base-cuda/Dockerfile [hunk 2]
```diff
@@ -108,15 +108,18 @@ RUN \
...
 
 RUN \
     # install DeepSpeed
-    pip install deepspeed==0.5.7
+    pip install deepspeed==0.5.7 && \
+    python -c "import deepspeed ;  print(deepspeed.__version__)"
```
Comment (carmocca) [inline dockers/base-cuda/Dockerfile:117]:
```suggestion
    python -c "import deepspeed; print(deepspeed.__version__)"
```

File: dockers/base-cuda/Dockerfile [hunk 3]
```diff
@@ -108,15 +108,18 @@ RUN \
...
 
 RUN \
     # Show what we have
     pip --version && \
     pip list && \
     python -c "import sys; ver = sys.version_info ; assert f'{ver.major}.{ver.minor}' == '$PYTHON_VERSION', ver" && \
-    python -c "import torch; assert torch.__version__.startswith('$PYTORCH_VERSION'), torch.__version__"
+    python -c "import torch; assert torch.__version__.startswith('$PYTORCH_VERSION'), torch.__version__" && \
```
Comment (carmocca) [inline dockers/base-cuda/Dockerfile:124]: This code is also here:

https://github.com/PyTorchLightning/pytorch-lightning/blob/32e7d32956e1685d36f2ab0ca3770baa2f76ce10/dockers/base-conda/Dockerfile#L132

https://github.com/PyTorchLightning/pytorch-lightning/blob/32e7d32956e1685d36f2ab0ca3770baa2f76ce10/dockers/base-ipu/Dockerfile#L97

You might want to update it too.

**Review by carmocca — APPROVED** [review]

File: dockers/base-cuda/Dockerfile [hunk 4]
```diff
@@ -108,15 +108,18 @@ RUN \
...
+    python -c "import deepspeed; print(deepspeed.__version__)"
 
 RUN \
     # Show what we have
     pip --version && \
     pip list && \
     python -c "import sys; ver = sys.version_info ; assert f'{ver.major}.{ver.minor}' == '$PYTHON_VERSION', ver" && \
-    python -c "import torch; assert torch.__version__.startswith('$PYTORCH_VERSION'), torch.__version__"
+    python -c "import torch; assert torch.__version__.startswith('$PYTORCH_VERSION'), torch.__version__" && \
+    python -c "import horovod.torch"
```
Comment (rohitgr7) [inline dockers/base-cuda/Dockerfile:125]: forgot to print the version? just like other sanity checks?

Comment (akihironitta) [reply dockers/base-cuda/Dockerfile:124]: The reason why we put the import statement here is just because we want the CI workflow to fail if importing `horovod.torch` isn't successful, so I don't think we need one here since the above `pip list` shows the version if installed.
